viadot/sources/outlook.py

```diff
@@ -90,100 +82,135 @@ def __init__(
+        """to retrieve all messages from all the mailboxes passed in the dictionary.
+
+        Args:
+            dict_folder (dict): Mailboxes dictionary holder, with the following structure:
+                "parent (sub)folder|(sub)folder": Mailbox.
+            limit (int, optional): Number of fetched top messages. Defaults to 10000.
+
+        Returns:
+            list: A list with all messages from all Mailboxes.
         """
         data = []
-        mailbox_generators_list = []
-        for m in self.mailbox_folders:
-            base_str = f"self.mailbox_obj.{m}_folder().get_messages({self.limit})"
-            mailbox_generators_list.append(eval(base_str))
-
-        for mailbox_generator in mailbox_generators_list:
-            while True:
-                try:
-                    message = next(mailbox_generator)
-                    received_time = message.received
-                    date_time_str = str(received_time)
-                    date_time_stripped = date_time_str[0:19]
-                    date_obj = datetime.datetime.strptime(
-                        date_time_stripped, "%Y-%m-%d %H:%M:%S"
-                    )
-                    if (
-                        date_obj < self.date_range_start_time
-                        or date_obj > self.date_range_end_time
-                    ):
-                        continue
+        for key, value in list(dict_folder.items()):
+            count = 0
+            for message in value.get_messages(limit=limit):
+                received_time = message.received
+                date_obj = datetime.fromisoformat(str(received_time))
+                if (
+                    self.date_range_start_time.replace(tzinfo=self.utc)
+                    < date_obj
+                    < self.date_range_end_time.replace(tzinfo=self.utc)
+                ):
+                    count += 1
+                    fetched = message.to_api_data()
+                    sender_mail = fetched.get("from", None)
+                    if sender_mail is not None:
+                        sender_mail = fetched["from"]["emailAddress"]["address"]
+                    recivers_list = fetched.get("toRecipients")
+                    recivers = " "
+                    if recivers_list is not None:
+                        recivers = ", ".join(
+                            reciver["emailAddress"]["address"]
+                            for reciver in recivers_list
+                        )
+                    categories = " "
+                    if message.categories is not None:
+                        categories = ", ".join(
+                            categories for categories in message.categories
+                        )
+                    conversation_index = " "
+                    if message.conversation_index is not None:
+                        conversation_index = message.conversation_index
+                    row = {
```

Comment:
Please check if it is possible to add also `subject` and `read` ('isRead') field

Reply:
sorry, but don't exactly understand what you want to point out.

Reply:
There are additional (new) fields. The question is whether there is a chance to add new fields/columns to the existing  Outlook extract.

Reply:
for now, the new paradigm does not require any other field, and adding more, I think, could be for a future big upgrade (if needed, because I don't think so) to choose from all avialable fields.